pymets/mets_structure.py

```diff
@@ -153,7 +153,7 @@ def create_xml_file(self, mets_filename, nsmap=None):
             f = open(mets_filename, 'w')
             f.write(self.create_xml_string(nsmap).encode("utf-8"))
```

Comment:
We should probably either open this file in `wb` mode since we're trying to write bytes, or we can save some space and just remove the `.encode("utf-8")` since we can write a normal str in `w` mode.

Reply:
As it currently is, I believe this will cause an error--well kind of multiple errors :). Calling `self.create_xml_string` at all is probably going to error because line 178:
```
  xml = '<?xml version="1.0" encoding="UTF-8"?>\n' + tostring(root, pretty_print=True)
```
will error. This is because `tostring` misleadingly actually returns a `bytes` object unless `encoding='unicode'`, so trying to concatenate a `str` and `bytes` will error (f there is not a test for this method, there should be). Then above, line 153 and 154 are also problematic because `encode` is for changing `str` to `bytes`, but since `self.create_xml_string()` would be returning `bytes` (if not erroring as it likely is now) we are then saying encode `bytes` to `bytes` which will error with `AttributeError: 'bytes' object has no attribute 'encode'`. Even if we could accomplish the `encode` we would error on the `write` because we would be trying to write `bytes` to a file that is expecting `str` to be written to it. What @somexpert recommends sounds good to me, change the `open` mode to `wb` so it is expecting `bytes`, then take off the `.encode("utf-8")`, and fix what is returned from `self.create_xml_string`. In pyuntl we changed such things to:
```
    return b'<?xml version="1.0" encoding="UTF-8"?>\n' + tostring(
        root,
        encoding='UTF-8',
        xml_declaration=False,
        pretty_print=True
    )
```
Note: you could have `tostring` add the xml declaration for you, but what we found is that it does it with single quotes and lowercase "utf-8" which was not consistent with the rest of the XML file or existing XML files, so we didn't.

Reply:
I gave this a shot and hopefully added a meaningful test based on your suggestion.